### PR TITLE
recovering optimized stability

### PR DESCRIPTION
**Pull request by magarciaEPFL**
Discussion in https://issues.scala-lang.org/browse/SI-5891 and also in https://issues.scala-lang.org/browse/SI-5442

review by @VladUreche

### CODE REVIEW

File: src/compiler/scala/tools/nsc/backend/opt/Inliners.scala [hunk 1]
```diff
@@ -12,6 +12,29 @@ import scala.tools.nsc.symtab._
...
+ *                (the accesibility of methods and constructors isn't touched by the inliner).
+ *
+ * Thus we add one more goal to our list:
+ *   (c) Compile C (either optimized or not) against any of L or L',
+ *       so that it runs with either L or L' (in particular, compile against L' and run with L).
+ *
+ * The chosen strategy is described in some detail in the comments for `accessRequirements()` and `potentiallyPublicized()`.
+ * Documentation at http://lamp.epfl.ch/~magarcia/ScalaCompilerCornerReloaded/2011Q4/Inliner.pdf
+ *
  *  @author Iulian Dragos
```
Comment (dragos) [inline src/compiler/scala/tools/nsc/backend/opt/Inliners.scala:38]: You should remove this author tag, or add your name :)

Comment (paulp) [inline src/compiler/scala/tools/nsc/backend/opt/Inliners.scala:38]: As a general comment, it's a thrill for me to see comments of this quality.  Not too high-level, not too low-level, and explaining what really needs explaining.  They're certainly better than any of mine; in fact they inspire me to do better with my own.

**Comment by magarciaEPFL** [comment]
PLS REBUILD ALL

**Comment by gkossakowski** [comment]
I just tested this change by rebuilding locker and I can testify it helps a lot with closure elimination. I think we should merge this ASAP.

**Pull request merged**